### PR TITLE
[SPARK-47178][PYTHON][TESTS] Add a test case for createDataFrame with dataclasses

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR adds a test case for dataclasses with createDataFrame that is already supported:

```python
from dataclasses import dataclass

@dataclass
class User:
    name: str
    age: int
    is_active: bool

user = User(name="John", age=30, is_active=True)
spark.createDataFrame([user]).show()
```

### Why are the changes needed?

To make sure dataclasses work with PySpark.

### Does this PR introduce _any_ user-facing change?

No, test-only

### How was this patch tested?

Added test case.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/pyspark/sql/session.py [hunk 1]
```diff
@@ -1299,7 +1299,7 @@ def createDataFrame(  # type: ignore[misc]
         ----------
         data : :class:`RDD` or iterable
             an RDD of any kind of SQL data representation (:class:`Row`,
-            :class:`tuple`, ``int``, ``boolean``, etc.), or :class:`list`,
+            :class:`tuple`, ``int``, ``boolean``, ``dict``` etc.), or :class:`list`,
```
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:1302]:
```suggestion
            :class:`tuple`, ``int``, ``boolean``, ``dict`` etc.), or :class:`list`,
```

File: python/pyspark/sql/session.py [hunk 2]
```diff
@@ -1299,7 +1299,7 @@ def createDataFrame(  # type: ignore[misc]
         ----------
         data : :class:`RDD` or iterable
             an RDD of any kind of SQL data representation (:class:`Row`,
-            :class:`tuple`, ``int``, ``boolean``, etc.), or :class:`list`,
+            :class:`tuple`, ``int``, ``boolean``, ``dict`` etc.), or :class:`list`,
```
Comment (HyukjinKwon) [inline python/pyspark/sql/session.py:1302]:
```suggestion
            :class:`tuple`, ``int``, ``boolean``, ``dict``, etc.), or :class:`list`,
```

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**